types/wordpress__notices/index.d.ts

```diff
@@ -63,4 +63,21 @@ export interface Options {
      * @defaultValue `'default'`
      */
     type: 'default' | 'snackbar';
+    /**
+     *  An icon displayed with the notice.
+     * @defaultValue `null`
+     */
+    icon: null | JSX.Element;
+    /**
+     * Whether the notice includes
+     * an explict dismiss button and
+     * can't be dismissed by clicking
+     * the body of the notice.
+     * @defaultValue `false`
+     */
+    explicitDismiss: boolean;
+    /**
+     *  Called when the notice is dismissed.
+     */
+    onDismiss?(): void;
 }
```

Comment:
> What's the purpose in exporting this options object?

Sorry, I don't know much about this. AFAIK, It's added in this PR (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/36700). 

> Should we have a few extra lines to define createNotice and its curried friends like createSuccessNotice?

Yep, we have them [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/44424ffb1ae5e383a9a2be6f91ad621c3758c740/types/wordpress__notices/store/actions.d.ts#L10-L50).


> We can remove the ? from onDismiss?: in the Options type too, since it's the same as every other property (optional) and then use Partial on the function type signature to indicate that optionality of all the options.

Good point. Removed in 3c9e1babc83892df095c60bb2b4e2d9301a07c12. Thanks!